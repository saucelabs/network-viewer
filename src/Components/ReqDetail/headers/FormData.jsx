import React from 'react';
import PropTypes from 'prop-types';

import Styles from './../Headers.styles.scss';

const FormData = ({ data, isPayloadTransformed }) => (
  <div className={Styles['header-detail']}>
    {data.headers.postData.params.map(({ name, value }, index) => (
      <p
        key={`${name}-${index}`}
        className={Styles['info-row']}
      >
        <span className={Styles['info-caption']}>
          {`${name}:`}
        </span>
        <span className={Styles['info-value']}>
          {isPayloadTransformed ? decodeURIComponent(value) : value}
        </span>
      </p>
    ))}
  </div>
);

FormData.propTypes = {
  data: PropTypes.object,
  isPayloadTransformed: PropTypes.bool.isRequired,
};

FormData.defaultProps = {
  data: null,
};

export default FormData;
