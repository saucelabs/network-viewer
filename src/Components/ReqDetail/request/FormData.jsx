import React from 'react';
import PropTypes from 'prop-types';

import Styles from '../ReqDetail.styles.scss';

const FormData = ({ data, isPayloadTransformed }) => (
  <div className={Styles['section-detail']}>
    {data.headers.postData.params.map(({ name, value }, index) => (
      <div
        key={`${name}-${index}`}
        className={Styles['info-row']}
      >
        <span className={Styles['info-caption']}>
          {`${name}:`}
        </span>
        <span className={Styles['info-value']}>
          {isPayloadTransformed ? decodeURIComponent(value) : value}
        </span>
      </div>
    ))}
  </div>
);

FormData.propTypes = {
  data: PropTypes.object,
  isPayloadTransformed: PropTypes.bool.isRequired,
};

FormData.defaultProps = {
  data: null,
};

export default FormData;
