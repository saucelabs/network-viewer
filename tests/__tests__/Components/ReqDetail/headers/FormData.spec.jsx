import React from 'react';
import { shallow } from 'enzyme';

import FormData from './../../../../../src/Components/ReqDetail/headers/FormData';

describe('FormData', () => {
  const props = {
    data: {
      headers: {
        postData: {
          params: [{
            name: 'foo',
            value: 'bar',
          }],
        },
      },
    },
    isURLEncoded: true,
  };

  it('renders without crashing', () => {
    const element = shallow(<FormData {...props} />);
    expect(element).toMatchSnapshot();
  });
});
