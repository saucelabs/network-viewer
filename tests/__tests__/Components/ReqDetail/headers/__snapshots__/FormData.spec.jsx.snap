// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormData renders without crashing 1`] = `
<div>
  <p
    key="foo-0"
  >
    <span>
      foo:
    </span>
    <span>
      bar
    </span>
  </p>
</div>
`;
